plugin/storage/cassandra/factory.go

```diff
@@ -232,6 +232,13 @@ var _ io.Closer = (*Factory)(nil)
 
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
+	if f.primarySession != nil {
+		f.primarySession.Close()
+	}
```

Comment:
```suggestion
	var errs []error
	if f.primarySession != nil {
		errs = append(errs, f.primarySession.Close())
	}
```
and at the end return `errors.join(errs...)`

Reply:
The `Close` function in this case does not return an error - I think the concrete implementation in this case comes from here: https://github.com/gocql/gocql/blob/1c9ec8f651a2f47e2ac7d7b7e5030b1a6d7a2575/session.go#L463.

Having said that there is existing code that further down which swallows an error, so I can address that.